src/bootstrap-table.js

```diff
@@ -1653,6 +1653,15 @@ class BootstrapTable {
       }
     }
 
+    if (this.options.search && this.options.sidePagination === 'server') {
+      params.searchable = []
+      for (const column of this.columns) {
+        if (!column.checkbox && column.searchable && ((this.options.visibleSearch && column.visible) || !this.options.visibleSearch)) {
+          params.searchable.push(column.field)
+        }
+      }
+    }
```

Comment:
https://live.bootstrap-table.com/code/wenzhixin/3069

I think that only sends the parameters to the server when sets a searchable `false` for certain columns.

```js
    if (this.options.search && this.options.sidePagination === 'server') {
      const searchable = []

      for (const column of this.columns) {
        if (!column.checkbox && column.searchable && ((this.options.visibleSearch && column.visible) || !this.options.visibleSearch)) {
         searchable.push(column.field)
        }
      }

      if (this.columns.filter(column => !column.searchable).length) {
        params.searchable = searchable
      }
    }
```

Reply:
I checked my and your example again, it sends the field to the server if searchable is set to `true`.

Reply:
Sorry for the late reply. I mean, if we don’t set searchable for all our columns, then we don’t send params to the server, which keeps the same behavior as before. What do you think?

Reply:
![image](https://user-images.githubusercontent.com/2117018/83087615-40d5a980-a0c4-11ea-8247-a684f583cbb7.png)

Reply:
But isnt it correct that we dont send the paramteres to the server if we dont have searchable columns ?
So the server can check if there are searchable columns, if yes filter the data based on the columns and the search string. If not ignore the search and return all data.

Reply:
The server can check if there are searchable columns, if yes filter the data based on the columns and the search string. If not filter the data based on the search string, which keeps the same behavior as before.

Reply:
No its not the same behavior as before.

Before the server only got the search string, which means the server didnt know for which columns he should search.

After the server get a list of columns which he should search for the given search string.
If the Server get no list of columns, the server can return the whole dataset.

@djhvscf could please also check tha ?

Reply:
![image](https://user-images.githubusercontent.com/2117018/83733993-f4b8d500-a680-11ea-9572-bb0fa7614be3.png)

In most cases, the searchable option of the column is not set, which means that all columns are searched. Without this parameter, it means **All**.

For example https://developer.github.com/v3/issues/#list-repository-issues.

> labels: A list of comma separated label names.

If there are 100 labels in our project, they are label1, label2, ... label100. When we want to search for issues with all labels, we don't need to write: labels=label1,label2...label100, we don't need to pass this parameter, it means all.